src/imgix-core-js.js

```diff
@@ -1,37 +1,32 @@
 (function (global, factory) {
   if (typeof define === 'function' && define.amd) {
-    define('Imgix', ['exports', 'md5', 'js-base64', 'crc'], factory);
+    define('Imgix', ['exports', 'md5', 'js-base64'], factory);
   } else if (typeof exports !== 'undefined') {
-    module.exports = factory(exports, require('md5'), require('js-base64').Base64, require('crc'));
+    module.exports = factory(exports, require('md5'), require('js-base64').Base64);
   } else {
     var mod = {
       exports: {}
     };
-    global.ImgixClient = factory(mod.exports, global.md5, global.Base64, global.crc);
+    global.ImgixClient = factory(mod.exports, global.md5, global.Base64);
   }
-})(this, function (exports, _md5, _jsBase64, _crc) {
+})(this, function (exports, _md5, _jsBase64) {
   var md5 = _md5;
   var Base64 = _jsBase64.Base64 || _jsBase64;
-  var crc = _crc;
 
   var VERSION = '1.3.0';
-  var SHARD_STRATEGY_CRC = 'crc';
-  var SHARD_STRATEGY_CYCLE = 'cycle';
   var DOMAIN_REGEX = /^(?:[a-z\d\-_]{1,62}\.){0,125}(?:[a-z\d](?:\-(?=\-*[a-z\d])|[a-z]|\d){0,62}\.)[a-z\d]{1,63}$/i;
   var DEFAULTS = {
     host: null,
-    domains: [],
+    domain: '',
```

Comment:
One thing that I'm a little iffy on: we mention in the last release that domain sharding would be deprecated but failed to mention this change to the `domains` field specifically. Unlike in [imgix-rb](https://github.com/imgix/imgix-rb/blob/be8b5d55517b107aa9373c964b6049855e6f842d/lib/imgix/client.rb#L17), we only accept a `domains` argument, which means that users who don't use sharding would still be affected by this release. Because of that, I wonder if this change would still be considered kosher.

A compromise could be deprecating (but still respecting) the `domains` field, while throwing a warning. If that's the case, I'm thinking might be cleaner to just pull that change out into a separate deprecation release before moving forward with this PR.

Reply:
I agree with the "compromise" approach. Releases are free, so we shouldn't be afraid of making as many releases as necessary to ensure a smooth ramp for our users.